lib/connection.js

```diff
@@ -24,7 +24,12 @@ function getSocket(url, options) {
     // If invalid auth, we will not try to reconnect.
     let invalidAuth = false;
 
-    const closeMessage = () => {
+    const closeMessage = (evt) => {
+      if (evt.target && evt.target.removeEventListener) {
```

Comment:
Why not use `socket` variable?

Reply:
That's a good point. Let's try to be consistent. We are using `socket` in the `handleMessage` listener and also depend on `triesLeft` from the outerscope. I think sticking with `socket` is fine.

Reply:
Done